flask-server/test_extraction.py

```diff
@@ -0,0 +1,65 @@
+import os
+import unittest
+from dotenv import load_dotenv
+from extraction import (
+    get_data_folders,
+    set_path,
+    get_gamma_options,
+    get_solution_name,
+    extract_text,
+    get_alternate_solutions,
+    get_gamma_data,
+)
+from expected_results import (
+    expected_folders,
+    expected_options,
+    expected_solution_filename,
+    expected_extracted_text_data,
+    expected_alt_folders,
+    expected_data,
+)
+
+
+class TestExtraction(unittest.TestCase):
+    def test_get_data_folders(self):
+        self.assertEqual(get_data_folders(), expected_folders)
+
+    def test_get_gamma_options(self):
+        load_dotenv()
+        path = os.path.join(os.getenv("MALTA_DATA_FOLDER"), str(os.getenv("TEST_DATA")))
+        self.assertEqual(get_gamma_options(path), expected_options)
+
+    def test_set_path(self):
+        gamma = 200
+        expected_path = f"gammas/{gamma}/"
+        self.assertEqual(set_path(gamma), expected_path)
+
+    def test_get_solution_name(self):
+        folder = str(os.getenv("TEST_DATA"))
+        self.assertEqual(
+            get_solution_name(folder, "_model_fit.pdf"), expected_solution_filename
+        )
+
+    def test_extract(self):
+        load_dotenv()
+        path = "gammas/100/"
+        folder = os.getenv("TEST_DATA")
+
+        self.assertEqual(extract_text(path, 0, folder), expected_extracted_text_data)
```

Comment:
This writes test output to the input directory, and leaves clutter behind after the test is run. Among other issues, this can cause tests to pass when they should fail, because they are checking output from previous, successful test runs. Much better to write to a temporary directory: https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

For example, see: https://github.com/oicr-gsi/djerba/blob/003c6227c0bb678b9afe1c9ab995b27d6a4fe4aa/src/lib/djerba/main.py#L145
(The `with` block ensures Python cleans up the directory after it is no longer in use.)